source/cbor/Cbor.cpp

```diff
@@ -0,0 +1,260 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+#include <aws/crt/Api.h>
+#include <aws/crt/cbor/Cbor.h>
+
+namespace Aws
+{
+    namespace Crt
+    {
+        namespace Cbor
+        {
+            /*****************************************************
+             *
+             * CborEncoder
+             *
+             *****************************************************/
+            CborEncoder::CborEncoder(Crt::Allocator *allocator) noexcept
+            {
+                m_encoder = aws_cbor_encoder_new(allocator);
```

Comment:
there's no possible way that aws_cbor_encoder_new() or aws_cbor_decoder_new() could ever return NULL in the future, right? right?

if so, we won't want to use normal C++ constructors, because they can't fail without the use of exceptions, which we turn off

Reply:
I don't think we will error let the new function fail for the encoder and decoder, besides of allocation failure